src/models/ScoreLog.php

```diff
@@ -69,28 +75,49 @@ private static function scorelogFromRow(Map<string, string> $row): ScoreLog {
     int $level_id,
     int $team_id,
     bool $any_team,
+    bool $refresh = false,
   ): Awaitable<bool> {
     $db = await self::genDb();
-
-    if ($any_team) {
-      $result =
-        await $db->queryf(
-          'SELECT COUNT(*) FROM scores_log WHERE level_id = %d AND team_id IN (SELECT id FROM teams WHERE id != %d AND visible = 1)',
-          $level_id,
-          $team_id,
-        );
-    } else {
-      $result =
-        await $db->queryf(
-          'SELECT COUNT(*) FROM scores_log WHERE level_id = %d AND team_id = %d',
-          $level_id,
-          $team_id,
-        );
+    $mc_result = self::getMCRecords('LEVEL_CAPTURES');
+    if ((!$mc_result) || (count($mc_result) === 0) || ($refresh)) {
+      $level_captures = Map {};
+      $result = await $db->queryf('SELECT level_id, team_id FROM scores_log');
+      foreach ($result->mapRows() as $row) {
+        if ($level_captures->contains(intval($row->get("level_id")))) {
+          $level_capture_teams =
+            $level_captures->get(intval($row->get("level_id")));
+          /* HH_IGNORE_ERROR[4064] */
```

Comment:
Oops.  Sorry about that.  It's the same suppression as found on line 109 (below this one).

Also, work on fixing tests seems to have been carried out in parallel, as #315 seems to be a duplicate of #309.  #315 can likely be closed if #309 fully resolves the test issues.